app/services/agency_identity_linker.rb

```diff
@@ -9,6 +9,20 @@ def link_identity
       AgencyIdentity.new(user_id: @sp_identity.user_id, uuid: @sp_identity.uuid)
   end
 
+  def self.for(user:, service_provider:)
+    agency = service_provider.agency
+
+    ai = AgencyIdentity.where(user: user, agency: agency).take
+    return ai if ai.present?
+
+    spi = ServiceProviderIdentity.where(
+      user: user, service_provider: service_provider.issuer,
+    ).take
+
+    return nil unless spi.present?
+    new(spi).link_identity
```

Comment:
I would love a version of this method that is read-only and doesn't attempt to link? But I'm not sure that would work here, for the AttemptsAPI.... since usually the ID is only created after an account is created + consented to? Maybe we put in a pin and circle back

Reply:
I was unclear on some of the existing semantics here. Just to be clear, are you suggesting to leave this as-is for now but consider circling back on this later? Or are you saying that this is will prematurely create these links and should be changed now?

Reply:
Sorry that was a mess of a comment. Let's leave this as-is